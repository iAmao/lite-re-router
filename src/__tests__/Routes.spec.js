import React from 'react';

import renderer from 'react-test-renderer';

import { shallow, mount, render } from 'enzyme';

import { Routes, Router } from '../';


describe('Routes', () => {
  describe('Snapshot Test', () => {
    it('should render the router component', () => {
      const tree = renderer.create(
        <Router render={() => {
          return <Routes routes={() => (
            [
              ['/', () => <h3>Home</h3>]
            ]
          )} />
        }} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('Integration Test', () => {
    it('should match component to route', () => {
      let wrapper = mount(<Router render={() => {
          return <Routes routes={() => (
            [
              ['/', () => <h3>Home</h3>],
              ['/about', () => <h3>About</h3>]
            ]
          )} />
        }} />);
      expect(wrapper.find('h3').text()).toEqual('Home');
      window.history.pushState(null, '', '/about');
      wrapper = mount(<Router render={() => {
          return <Routes routes={() => (
            [
              ['/', () => <h3>Home</h3>],
              ['/about', () => <h3>About</h3>]
            ]
          )} />
        }} />);
      expect(wrapper.find('h3').text()).toEqual('About');
    });

    it('should match component with route param', () => {
      window.history.pushState(null, '', '/about/me');
      let wrapper = mount(<Router render={() => {
          return <Routes routes={() => (
            [
              ['/about/:id', (props) => <h3>About {props.location.params.id}</h3>],
            ]
          )} />
        }} />);
      expect(wrapper.find('h3').text()).toEqual('About me');
    });

    it('should match component with exact rote if specified', () => {
      window.history.pushState(null, '', '/about/me');
      let wrapper = mount(<Router render={() => {
          return <Routes routes={() => (
            [
              ['/about/:id', (props) => <h3>About {props.location.params.id}</h3>],
              ['/about/me', (props) => <h3>About Me</h3>],
              ['/about/edit/:id', (props) => <h3>Edit about Me</h3>]
            ]
          )} />
        }} />);
      expect(wrapper.find('h3').text()).toEqual('About Me');
    });

    it('should return a default 404 page if a route is not found', () => {
      window.history.pushState(null, '', '/about/me');
      let wrapper = mount(<Router render={() => {
          return <Routes routes={() => (
            [
              ['/', () => <h3>Home</h3>],
            ]
          )} />
        }} />);
      expect(wrapper.find('h2').text()).toEqual('404!');
    });

    it('should return a 404 page if a route is not found', () => {
      window.history.pushState(null, '', '/about/me');
      let wrapper = mount(<Router render={() => {
          return <Routes routes={() => (
            [
              ['/', () => <h3>Home</h3>],
              ['*', () => <h3>404</h3>],
            ]
          )} />
        }} />);
      expect(wrapper.find('h3').text()).toEqual('404');
    });
  });
});
