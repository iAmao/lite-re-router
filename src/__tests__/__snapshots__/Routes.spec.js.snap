// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes Snapshot Test should render the router component 1`] = `
<h3>
  Home
</h3>
`;
